presto-main/src/test/java/com/facebook/presto/operator/BenchmarkHashBuildAndJoinOperators.java

```diff
@@ -0,0 +1,316 @@
+
+        @Param({"0.1", "1", "2"})
+        protected double matchRate = 1;
+
+        protected List<Page> probePages;
+
+        @Setup
+        public void setup()
+        {
+            super.setup();
+            initializeProbePages();
+        }
+        public List<Page> getProbePages()
+        {
+            return probePages;
+        }
+
+        protected void initializeProbePages()
+        {
+            RowPagesBuilder probePagesBuilder = rowPagesBuilder(buildHashEnabled, hashChannels, ImmutableList.of(VARCHAR, BIGINT, BIGINT));
+
+            Random random = new Random(42);
+            int remainingRows = PROBE_ROWS_NUMBER;
+            int rowsInPage = 0;
+            while (remainingRows > 0) {
+                double roll = random.nextDouble();
+
+                int columnA = 20 + remainingRows;
+                int columnB = 30 + remainingRows;
+                int columnC = 40 + remainingRows;
+
+                int rowsCount = 1;
+                if (matchRate < 1) {
+                    // each row has matchRate chance to join
+                    if (roll > matchRate) {
+                        // generate not matched row
+                        columnA *= -1;
+                        columnB *= -1;
+                        columnC *= -1;
+                    }
+                }
+                else if (matchRate > 1) {
+                    // each row has will be repeated between one and 2*matchRate times
+                    roll = roll * 2 * matchRate + 1;
+                    // example for matchRate == 2:
+                    // roll is within [0, 5) range
+                    // rowsCount is within [0, 4] range, where each value has same probability
+                    // so expected rowsCount is 2
+                    rowsCount = (int) Math.floor(roll);
+                }
+
+                for (int i = 0; i < rowsCount; i++) {
+                    if (rowsInPage >= ROWS_PER_PAGE) {
+                        probePagesBuilder.pageBreak();
+                        rowsInPage = 0;
+                    }
+                    probePagesBuilder.row(format("%d", columnA), columnB, columnC);
+                    --remainingRows;
+                    rowsInPage++;
+                }
+            }
+            probePages = probePagesBuilder.build();
+        }
+    }
+
+    @Benchmark
+    public LookupSourceSupplier benchmarkBuildHash(BuildContext buildContext)
+    {
+        DriverContext driverContext = buildContext.createTaskContext().addPipelineContext(true, true).addDriverContext();
+
+        HashBuilderOperatorFactory hashBuilderOperatorFactory = new HashBuilderOperatorFactory(
+                HASH_BUILD_OPERATOR_ID,
+                TEST_PLAN_NODE_ID,
+                buildContext.getTypes(),
+                ImmutableMap.of(),
+                buildContext.getHashChannels(),
+                buildContext.getHashChannel(),
+                false,
+                Optional.empty(),
```

Comment:
? 

I am passing `buildContext.getHashChannel()` to the builder. `Optional.empty()` that you commented refers to `Optional<JoinFilterFunction> filterFunction` argument. Or am I missing something?